### PR TITLE
Chore: TIP-HT Dashboard Changes 

### PR DESCRIPTION
**Pull request by salehcerp**
### Fixed 

- changed levels to progress
- show null instead of NaN in HT Dashboard

### CODE REVIEW

File: web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx [hunk 1]
```diff
@@ -225,64 +228,34 @@ const HTSummative = () => {
...
+									<div className="text-xs text-gray-900 place-self-start self-center pl-2">
+										{gradeToColorMapper[k]} Group
+									</div>
+									{TIPSubjects.map(subject => {
+										const percentage = getPercentageFromMarks(
+											item?.[subject]?.obt,
+											item?.[subject]?.total
+										)
+										return (
+											<div className="">
```
Comment (hmzahmed) [inline web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx:245]: What purpose is this div serving?

File: web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx [hunk 2]
```diff
@@ -225,64 +228,34 @@ const HTSummative = () => {
...
+												<button
+													className={clsx(
+														'text-xs text-center shadow-md flex justify-center items-center rounded-md  w-12 h-10',
+														getColorClassFromPercent(percentage)
+													)}>
+													{!isNaN(percentage) ? percentage + '%' : ''}
+												</button>
```
Comment (hmzahmed) [inline web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx:252]: This UI element is not doing any action nor does it need any click functionality, we can change this from button to div

File: web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx [hunk 3]
```diff
@@ -225,64 +228,34 @@ const HTSummative = () => {
...
-													item?.['Maths']?.total
-												)
-											)
-										)}>
-										{getPercentageFromMarks(
-											item?.['Maths']?.obt,
-											item?.['Maths']?.total
-										) + '%'}
-									</button>
+						{Object.entries(items).map(([k, item], index) => {
```
Comment (hmzahmed) [inline web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx:231]: you can rename the variable 'k' to 'grade', we should always name variables something descriptive, 'item' can also be renamed to 'result'

File: web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx [hunk 4]
```diff
@@ -225,64 +228,34 @@ const HTSummative = () => {
...
+							return (
+								<div
+									key={index}
+									className="grid grid-cols-4 my-2 justify-items-center">
```
Comment (hmzahmed) [inline web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx:235]: We should avoid using index as key if we can, here we can use grade as index

File: web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx (hunk 2 above)
Comment (hmzahmed) [inline web/src/modules/TIP/HTDashboard/Tests/SummativeTest/index.tsx:252]: We should also provide a key whenever we render a list of elements, here we can provide a key by concatenating grade and subject

**Review by hmzahmed — CHANGES_REQUESTED** [review]
Congrats on your first PR, Nice work!! everything looks good. Just a few suggestions

**Review by mudssrali — APPROVED** [review]
Looks good to me. Thanks @salehcerp

**Pull request closed without merging**